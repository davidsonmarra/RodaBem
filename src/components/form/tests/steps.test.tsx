import React from 'react';
import {create} from 'react-test-renderer';
import Steps, {Step, Props} from '../steps';

const createComponent = ({...rest}: Props) => create(<Steps {...rest} />);

describe('Steps', () => {
  it('renders correctly', () => {
    const tree = createComponent({
      currentStep: 0,
      totalSteps: 3,
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders number of steps correctly', () => {
    const instance = createComponent({
      currentStep: 0,
      totalSteps: 3,
    }).root;
    expect(instance.findAllByType(Step).length).toBe(3);
  });

  it('renders active step correctly', () => {
    const instance = createComponent({
      currentStep: 1,
      totalSteps: 3,
    }).root;
    const steps = instance.findAllByType(Step);
    expect(steps[0].props.stepType).toBe('active');
    expect(steps[1].props.stepType).toBe('active');
    expect(steps[2].props.stepType).toBe('inactive');
  });

  it('renders no inactive step correctly', () => {
    const instance = createComponent({
      currentStep: 2,
      totalSteps: 3,
    }).root;
    const steps = instance.findAllByType(Step);
    expect(steps[0].props.stepType).toBe('active');
    expect(steps[1].props.stepType).toBe('active');
    expect(steps[2].props.stepType).toBe('active');
  });
});
