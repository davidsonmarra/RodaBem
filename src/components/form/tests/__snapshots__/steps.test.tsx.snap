// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#E30B0B",
        "flex": 1,
        "height": 12,
      }
    }
  />
  <View
    style={
      {
        "width": 8,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#ABABAB",
        "flex": 1,
        "height": 12,
      }
    }
  />
  <View
    style={
      {
        "width": 8,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#ABABAB",
        "flex": 1,
        "height": 12,
      }
    }
  />
  <View
    style={
      {
        "width": 8,
      }
    }
  />
</View>
`;
